velox/exec/AggregateCompanionAdapter.cpp

```diff
@@ -60,6 +60,13 @@ void AggregateCompanionFunctionBase::clearInternal() {
   fn_->clear();
 }
 
+void AggregateCompanionFunctionBase::initialize(
+    const std::vector<TypePtr>& rawInputType,
+    const facebook::velox::TypePtr& resultType,
+    const std::vector<VectorPtr>& constantInputs) {
+  fn_->initialize(rawInputType, resultType, constantInputs);
+}
```

Comment:
I realize that for companion functions, their initialize() method call would receive the current input types as the rawInputType. E.g., for foo_merge_extract, its initialize() methods receives the original intermediate type of foo as the rawInputType. Since all companion functions use the initialize() method of the main aggregation function, it would be helpful to let the initialize() method also receive a core::AggregationNode::Step to tell which step the rawInputType corresponds to.

Reply:
Yes, the `core::AggregationNode::Step` is also a piece of info that may be useful in the future. In some cases, we can determine the step of the current companion function by checking whether `rawInputType` is a ROW. However, having the `core::AggregationNode::Step` would be more suitable for all scenarios.